h2o-r/demos/rdemo.coxph.R

```diff
@@ -0,0 +1,86 @@
+# Load the libraries used to analyze the data
+library(survival)
+library(MASS)
+library(h2o)
+
+# Start H2O cluster
+h2o.init()
+
+# Get path to data file
+locate_source <- function(file) {
+  file_path <- try(h2o:::.h2o.locate(file), silent = TRUE)
+  if (inherits(file_path, "try-error")) {
+    file_path <- paste0("s3n://h2o-public-test-data/", file)
+  }
+  file_path
+}
+
+churn_path <- locate_source("smalldata/demos/WA_Fn-UseC_-Telco-Customer-Churn.csv")
+
+# Read data into H2O cluster
+churn_hex <- h2o.importFile(churn_path)
+
+# Use functionality from survival and MASS to select the predictors
+churn_df <- as.data.frame(churn_hex)
+
+r_model <-
+  stepAIC(
+    coxph(Surv(tenure, Churn == "Yes") ~
+            (I(gender == "Female") +
+               SeniorCitizen +
+               I(Partner == "Yes") +
+               I(Dependents == "Yes") +
+               (I(MultipleLines == "No") + I(MultipleLines == "Yes")) +
+               I(PaperlessBilling == "Yes") +
+               PaymentMethod +
+               MonthlyCharges) +
+            I(InternetService == "Yes") *
+            (I(OnlineSecurity == "Yes") +
+               I(OnlineBackup == "Yes") +
+               I(DeviceProtection == "Yes") +
+               I(TechSupport == "Yes") +
+               I(StreamingTV == "Yes") +
+               I(StreamingMovies == "Yes")) +
+            strata(Contract),
+          data = churn_df,
+          ties = "efron"),
+    k = log(sum(churn_df$Churn == "Yes"))
+  )
+
+r_model <-
+  update(r_model,
+         . ~ . - PaymentMethod +
+           I(PaymentMethod %in% c("Bank transfer (automatic)",
+                                  "Credit card (automatic)")))
+
+# Transform data for the H2O-based model
+churn_hex$HasChurned <- h2o.asnumeric(churn_hex$Churn == "Yes")
```

Comment:
Is the `asnumeric` really necessary?

Reply:
Removed

Reply:
I think all the other calls to `h2o.asnumeric` are redundant as well. It is not a big deal but might look more readable without it.